docs/content/topics/scripting/guide.md

```diff
@@ -9,1401 +9,867 @@ aliases:
 
-If we look at the output of the pod, we'll see this:
+Created event "b45720c4-115c-4c9b-b668-a872479f2210".
 
+Waiting for event's worker to be RUNNING...
+2021-09-23T16:30:29.267Z INFO: brigade-worker version: 927850b-dirty
+Project: first-payload
+Hello, Brigade!
 ```
-$ kubectl logs echo-1504074306432-master
-Project github.com/brigadecore/empty-testbed
-Event exec
-```
-
-At this point we have seen how we can access information about the project and event. In
-the next section we are going to turn to storage. We are going to see how Brigade provides
-ways for builds and jobs to share storage space.
-
-## Storing Data with Caches and Shared Space
-
-There are many ways that developers can store and retrieve data from Brigade. For example,
-object storage systems like Azure Object Storate or Amazon S3 or hosted database
-providers. Brigade developers may choose to use those tools from within jobs.
-
-But Brigade comes with a few built-in options that are useful in writing basic Brigade builds,
-and which don't require modifying your containers. In this part of the guide, we cover two
-built-in shared directories.
-
-### Build Storage (Shared Space)
 
-Each build gets its own shared storage. This is a file path that can be accessed by every
-job during the build, but which does not survive after the build has completed.
+Note that some event and project data should be treated with care. Things like
+`event.project.secrets` or event `event.worker.git.cloneURL` might not be the
+sorts of information you want accidentally displayed. Check out the [Secrets]
+guide for examples on how to safely handle project secrets in your scripts.
 
-When enabled, storage is mounted to `/mnt/brigade/share` on each job's container.
+[Secrets]: /topics/project-developers/secrets
 
-```javascript
-const { events, Job, Group } = require("brigadier")
-
-events.on("exec", (e, p) => {
-  var dest = "/mnt/brigade/share/hello.txt"
-  var one = new Job("one", "alpine:3.4", ["echo hello > " + dest])
-  var two = new Job("two", "alpine:3.4", ["echo world >> " + dest])
-  var three = new Job("three", "alpine:3.4", ["cat " + dest])
+## Worker storage and shared workspace
 
-  one.storage.enabled = true
-  two.storage.enabled = true
-  three.storage.enabled = true
+Brigade offers the ability to set up storage for the worker that can then be
+shared amongst jobs. This functionality isn't enabled by default and needs to
+be configured on the `workerTemplate` section of the project definiton as well
+as on each job in the script that requires access to the workspace.
 
-  Group.runEach([one, two, three])
-})
-```
-[brigade-15.js](../../examples/brigade-15.js)
+As background, the decision to make this functionality opt-in as opposed to
+opt-out was based on the desire to have job results be cache-able and jobs
+themselves resumable/retryable, which can currently only be accomplished when
+they do not share a workspace.
```

Comment:
I'm iffy on whether we should include this. You and I spent so much time on this that we understand this well. I'm trying to look at this through the eyes of a new user and I'm not sure I understood that.

Maybe it's better to cut this detail than to expand further? Honestly, I really don't know. Just thinking out loud.

Reply:
Will cut; can re-intro when/if it proves necessary.